core/src/tasks/system/status/status.ts

```diff
@@ -38,8 +38,10 @@ export class StatusTask extends CLSTask {
       const complete = await this.checkForComplete(samples);
 
       if (runMode === "cli:run" && complete) {
-        await this.logFinalSummary();
+        const sweepProfiles: Task = api.tasks.tasks["profiles:sweep"];
+        await sweepProfiles.run({}, {});
```

Comment:
I'm not super convinced this is the right place to put this... suggestions welcome!

Reply:
It's a little weird to have one task call another.  This might be a signal to move your logic into an Op - `const deletedCount = ProfileOps.sweep(limit = 1000)` or similar, so you can call it directly here.  
* You also may need to call this method more than once, if there are a lot of profiles to remove... so maybe the method returns how many profiles it found and you keep looping until it's done.

Reply:
An alternative strategy:
* You could modify `checkForComplete` below, and include a check for Profiles that have a null  directly mapped property, and not let the server stop if there are any to delete
* If there are any Profiles that should be deleted, `await task.enqueue('profiles:sweep')` and let it process out of band

Reply:
I agree!
I went with a bit of a combination of both options. 

Something to note is that since we're now waiting for these profiles to be deleted, the task now runs more frequently so that `roo run` completes in a reasonable time. 30s seems okay to me, specially because we're limiting the amount of profiles each time it runs.